pilot/pkg/networking/core/v1alpha3/accesslog.go

```diff
@@ -127,9 +127,9 @@ func newAccessLogBuilder() *AccessLogBuilder {
 
 func (b *AccessLogBuilder) setTCPAccessLog(push *model.PushContext, proxy *model.Proxy, tcp *tcp.TcpProxy, class networking.ListenerClass) {
 	mesh := push.Mesh
-	cfg := push.Telemetry.AccessLogging(push, proxy, class)
+	cfgs := push.Telemetry.AccessLogging(push, proxy, class)
 
-	if cfg == nil {
+	if cfgs == nil {
```

Comment:
curious: should we also check `len(cfgs)`? naively, it seems like `len(cfgs) == 0` might be appropriate (here and below) and cover the `nil` case too.

Reply:
we can improve this later, but for now keep the behavior as before, more details can be found in following test case:
```golang
                {
			"empty",
			nil,
			networking.ListenerClassSidecarOutbound,
			sidecar,
			nil,
			nil, // No Telemetry API configured, fall back to legacy mesh config setting
		},
```